server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import static org.apache.james.mailbox.MailboxManager.MailboxSearchFetchType.Minimal;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.Mailbox;
+import org.apache.james.mailbox.model.MailboxMetaData;
+import org.apache.james.mailbox.model.search.MailboxQuery;
+import org.apache.james.user.api.UsersRepositoryException;
+
+import com.google.common.collect.ImmutableList;
+
+@Singleton
+public class UserMailboxesService {
+    private final MailboxManager mailboxManager;
+
+    @Inject
+    public UserMailboxesService(MailboxManager mailboxManager) {
+        this.mailboxManager = mailboxManager;
+    }
+
+    public List<Mailbox> listMailboxes(Username username) throws UsersRepositoryException {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(username);
+        try {
+            return listUserMailboxes(mailboxSession)
+                .map(MailboxMetaData::getMailbox)
+                .collect(ImmutableList.toImmutableList());
```

Comment:
I have seen this pattern where `toImmutableList` is called instead of `toList` in other parts of the codebase too, is there a particular reason why? `toList` returns an immutable list too.

Reply:
> toList returns an immutable list too.

Where did we get this statement / believe from?

![image](https://github.com/user-attachments/assets/dca8fc59-825f-4f1e-9601-735080295116)

looks like it is an arraylist... The finisher is `Collectors.castingIdentity()`...

Reply:
True, but calling toList() directly without collect returns an unmodifiable list according to the docs. 

But the implementation doesn't look efficient

![image](https://github.com/user-attachments/assets/b3480bb0-2bc9-4fea-8779-5a92e79970ca)

Reply:
either works for me :))
I am not sure which one is better

Reply:
I'm fine with either too, just asked to clarify if there was another reason like compatibility or better performance.

Reply:
"unmodifiable do not mean immutable. unmodifiable is merely some syntactic sugar that do not make things safe"

Quotes from Guava advocates.

Reply:
Some time, the simple reason just is `.toList` support from java 16, (we upgraded James to java 17 last year)
Some people are familar with old Guava syntax. 
:D just for fun